### ISSUE TITLE
Inward/Outward Polygon Offsetting

### ISSUE THREAD

**Issue by mcontim** [JuliaGeometry/Meshes.jl#1013]
Hi!

I'm investigating on how to produce an offsetted version of a polygon, both inward or outward, like in these StackOverflow threads:
- https://stackoverflow.com/questions/54033808/how-to-offset-polygon-edges
- https://stackoverflow.com/questions/1109536/an-algorithm-for-inflating-deflating-offsetting-buffering-polygons

I was wondering if there is anything already avalable in `Meshes.jl` to solve this problem. Thank you for the help!

**Comment by juliohm** [comment]
Hi @mcontim , we have something similar in the `Repair(10)` transform. You can try to `poly |> Repair(10)` to see if it works for now. What we really need is a better clipping algorithm (see #999) to create buffers, intersections, unions, etc.

**Comment by mcontim** [comment]
Hi @juliohm, thank you for the quick reply!

I had a look at `Repair(10)` I think it uses `Stretch` so the transformation is applied for each point of the polygon instead of offsetting each edge of the polygon (please correct me if I'm wrong).

In order to preserve the shape of the polygon, I'm looking for a solution to offset each edge of the polygon (like in the links above). I will try an implementation, then, if of interest, I can share the code and see if it is useful for `Meshes.jl`.

EDIT: I just realized there is a wrapper of the C library `Clipper` in JuliaGeometry, I will also have a look at that. You're probably already working to a full Julia implementation of i for `Meshes.jl` :D

**Comment by juliohm** [comment]
It is definitely useful. Did you have a chance to take a look into #979 ?

**Comment by mcontim** [comment]
Yes, I started working on it but unfortunately I had to put it in stand-by becuse of incoming deadlines with work.

I'm taking inspiration from `HomogeneousSampling`, I defined an ad-hoc struct for Fibonacci sampling (for the moment it will be just one, but I think we could define an abstract type then different sub-types of Fibonacci sampling for different sampling approaches).

Then the `sample` method should be only defined for `Disk`, `Ball`, and probably `Plane` (`CylinderSurface` would also make sense too, we can discuss it).
